PSModules/AzureApiFunctions.psm1

```diff
@@ -0,0 +1,60 @@
+function ApiRequest {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $Url,
+        [Parameter(Mandatory=$true)]
+        [string] $ApiKey,
+        [Parameter(Mandatory=$false)]
+        [string] $Method = "GET",
+        [Parameter(Mandatory=$false)]
+        [string] $ApiVersion = "2017-11-11",
+        [Parameter(Mandatory=$false)]
+        $Body
+    )
+
+<#
+.SYNOPSIS
+JSON invoke web request for Azure Search
+
+.DESCRIPTION
+Invokes a web request to Azure Search REST API assuming a JSON document
+
+.PARAMETER Url
+Web URL
+
+.PARAMETER ApiKey
+API key to access
+
+.PARAMETER Method
+Optional HTTP method, defaults to GET
+
+.PARAMETER ApiVersion
+Optional, override the API version, defaults to 2017-11-11
+
+.PARAMETER Body
+Optional hash to send as the body
+
+.EXAMPLE
+ApiRequest -SearchName dfc-foo-sch -ResourceGroupName dfc-foo-rg -IndexFilePath C:\path\to\index.json
+#>
+
+    $ApiHeaders = @{
+        "Content-Type" = "application/json"
+        "api-key"      = $ApiKey
+    }
+    $FullUrl = "$($Url)?api-version=$ApiVersion"
+    
+    if ($Body) {
+        $JsonBody = $Body | ConvertTo-Json
+        Write-Verbose "Body - $Body"
+        $WebResponse = Invoke-WebRequest -Uri $FullUrl -Method $Method -Headers $ApiHeaders -Body $JsonBody -UseBasicParsing
+    }
+    else {
+        $WebResponse = Invoke-WebRequest -Uri $FullUrl -Method $Method -Headers $ApiHeaders -UseBasicParsing
+    }
+    
+    if ($WebResponse.StatusCode -eq 200) {
+        return $WebResponse.Content | ConvertFrom-Json

```

Comment:
Is $WebResponse.Content well understood?  If it's unknown and could be a very complex document you might find the default depth for ConvertFrom-Json is to small which will give misleading results.  Might be worth setting it to something quite high say 10 or 20

Reply:
Good idea. I've not seen a depth higher than about 3 but I've changed this to a depth of 10